### PR TITLE
Fix NFKD for accented digraph followed by accent

### PR DESCRIPTION
**Pull request by hsivonen**
Fixes https://github.com/unicode-org/icu4x/discussions/4527

### CODE REVIEW

**Comment by sffc** [comment]
Please add an entry for this to CHANGELOG.md

File: components/normalizer/tests/tests.rs [hunk 1]
```diff
@@ -1308,6 +1308,19 @@ fn test_utf16_basic() {
...
+    assert_eq!(
+        normalizer.normalize("\u{01C4}\u{0323}"),
+        "DZ\u{0323}\u{030C}"
+    );
+    assert_eq!(
+        normalizer.normalize("DZ\u{030C}\u{0323}"),
+        "DZ\u{0323}\u{030C}"
+    );
+}
+
```
Comment (eggrobin) [inline components/normalizer/tests/tests.rs:1323]: This issue also affects NFD; please add the following test:
```rust
    let normalizer: DecomposingNormalizer = DecomposingNormalizer::new_nfd();
    assert_eq!(normalizer.normalize("\u{0DDD}\u{0334}"), "\u{0DD9}\u{0DCF}\u{0334}\u{0DCA}");
```

**Review by eggrobin — CHANGES_REQUESTED** [review]

File: components/normalizer/tests/tests.rs (hunk 1 above)
Comment (hsivonen) [reply components/normalizer/tests/tests.rs:1323]: Done. Thanks. (This character took a lot of attention on the collator side.)

**Comment by hsivonen** [comment]
> CHANGELOG.md

Done.

**Review by eggrobin — APPROVED** [review]

**Pull request merged**